### ISSUE TITLE
Sonoff ns panel support

### ISSUE THREAD

**Issue by Nailik** [Nailik/rhasspy_mobile#363]
because the sonoff ns panel runs android and has both a microphone and a speaker it might be very cool to support this device.

All features need to be checked, mostly:

- ui
- widget
- background services
- permissions
- file selection

**Comment by Nailik** [comment]
![PXL_20230811_214923374~2.jpg](https://github.com/Nailik/rhasspy_mobile/assets/13292441/a87566a0-46bf-4346-9d50-351310e462a0)

**Comment by dm-chechik** [comment]
@Nailik does it require additional components to be installed onto the device?

**Comment by Nailik** [comment]
@dm-chechik you need to enabled dev settings and i also installed another launcher in order to open the app (you could do this by adb bridge as well)
I used this https://blakadder.com/nspanel-pro-sideload/ (note since sonoff version 1.3.2 you can enable dev settins via their app and therefrore don't need to dissasamble the device)

**Comment by dm-chechik** [comment]
@Nailik thanks, I successfully installed the app on the NS Panel.
Regarding the microphone recording level, I have a similar issue on the Lenovo Smart Clock 2. It works however the recorded sound level is really silent. 
BTW the Google Assistant records sound very well. Probably they use some kind of `AutomaticGainControl` or what ever.

**Comment by Nailik** [comment]
I'll implement it for testing wich mic input Domain for Rhasspy3.
Docs https://developer.android.com/reference/android/media/audiofx/AutomaticGainControl

**Comment by dawid124** [comment]
@Nailik I have the same issue on my ns panel. Have you already tested AutomaticGainControl?

**Comment by Nailik** [comment]
> @Nailik I have the same issue on my ns panel. But when I test microphone on any record app, voice is much louder. Have you already tested AutomaticGainControl?

Yes but `isAvailable` returned false - also it returned false on my Pixel 7 Pro so i think it might be deprecated?

**Comment by dawid124** [comment]
Perhaps... I will test the recording functions in Android (on NS panel pro) and I hope that I will come back with some conclusions

I still have an idea for something that might be useful.
When I tested panel I connected it to a 'speakerphone' and the microphone source was from the panel
An option to change the source might also be useful.
And by the way, I really appreciate this app! :) thanks to it I will be able to really use the potential of rhasspy

**Comment by Nailik** [comment]
@dawid124 i created an issue for your idea https://github.com/Nailik/rhasspy_mobile/issues/527

Did you test any open source record app? Then i could look up the code they used to record audio.

**Comment by dawid124** [comment]
I just test code from:
https://developer.android.com/media/platform/mediarecorder#sample-code
app works fine, micophone volume is also fine, but i plan to check the increase vlume option from stackoverflow also. Now I try configure kdoctor on my windows :/

the first difference I see is that you use
.setAudioSource(MediaRecorder.AudioSource.VOICE_COMMUNICATION)

in example is:
setAudioSource(MediaRecorder.AudioSource.MIC)

**Comment by dawid124** [comment]
I  tested example from developer.android.com with VOICE_COMMUNICATION option and it is the issue. Voice is very quiet now.

Can I ask you for compile the apk with the change an MediaRecorder.AudioSource.MIC and share or send it to me?

**Comment by Nailik** [comment]
[rhasspy_mobile_V_0.5.6-147-arm64-v8a-release.apk.zip](https://github.com/Nailik/rhasspy_mobile/files/14013815/rhasspy_mobile_V_0.5.6-147-arm64-v8a-release.apk.zip)

Here you go, ps the file (is zipped) just shy of the 25MB limit of GitHub :D

**Comment by dawid124** [comment]
Thanks! Voice is ok now. So I'm waiting for the stable version :)

**Comment by Nailik** [comment]
i am going to add a select option - will need some days

**Comment by dawid124** [comment]
no rush, just let me know when will be done :)

**Comment by Nailik** [comment]
@dawid124 you can select the audio source in the latest nightly build
